### PR TITLE
Add test job to pull request

### PR DESCRIPTION
**Pull request by jakeberesford-palmetto**
## Pull request checklist

Before submitting your PR, please review the following checklist:

- [x] Consider adding a unit test if your PR resolves an issue.
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

### Breaking changes

- [ ] Check if this pull request introduces a breaking change

## What does this implement/fix? Explain your changes.

* Add test step to pull request workflow
* Attempt to test multiple python versions in CI

## Any other comments?
Not sure whether this will work... need to see what the action does in CI!

### CODE REVIEW

**Review by emekdahl-palmetto — APPROVED** [review]
this is great, thanks!

**Review by ethanknox-palmetto — APPROVED** [review]
if this works, this is awesome

**Comment by ethanknox-palmetto** [comment]
I think it's that install gnarlyness where you need to `python3 -m pip install .` with each one

File: .github/workflows/pull-request.yaml [hunk 1]
```diff
@@ -7,6 +7,24 @@ on:
...
+          - 3.7
+          - 3.8
+          - 3.9
+    container: python:${{ matrix.python-version }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install
+        run: |
+          pip install -r dev-requirements.txt 
+          python3 -m pip install .
```
Comment (jakeberesford-palmetto) [inline .github/workflows/pull-request.yaml:25]: @ethanknox-palmetto This does not feel right, but untangling the imports will also be a nightmare. Maybe the least-worst option for now?

**Comment by jakeberesford-palmetto** [comment]
> This is a correct fail! I used a walrus, not backwards compatible :(

Bad walrus.

I think we can merge this and fix the failure in a separate PR? Unless you want to tack a commit on to here?

**Comment by ethanknox-palmetto** [comment]
fix here for those last 2 pesky versions https://github.com/palmetto/palm-dbt/pull/37

File: .github/workflows/pull-request.yaml (hunk 1 above)
Comment (ethanknox-palmetto) [reply .github/workflows/pull-request.yaml:29]: nope this is absolutely necessary current-state. we will need to re-name a bunch of things to do relative imports and then get palm's plugin install to play with it, not something I'm into atm personally.

**Pull request merged**